awscrt/auth.py

```diff
@@ -255,6 +255,32 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
```

Comment:
there are a bunch of comments in the `AwsCredentialsProviderBase` class about how we don't support custom subclasses. We can remove them now. In fact, there's no need anymore for that base class anymore. For backwards compatibility, we can't remove its existence, but maybe we can remove the docstrings so it's omitted from the documentation.

I'd originally thought we'd have python programmers subclass `AwsCredentialsProviderBase` to make custom providers. But there are complex refcounting issues with that approach, the way you did it with the separate delegate object is nice and simple and graceful.